### ISSUE TITLE
Dashboard tries to download JAR even for a gRPC POM artifact

### ISSUE THREAD

**Issue by suztomo** [GoogleCloudPlatform/cloud-opensource-java#943]
Followup of https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/942#discussion_r329631742

```
suztomo@suxtomo24:~/cloud-opensource-java/dashboard$  mvn exec:java -Dexec.arguments="-f ../boms/cloud-oss-bom/pom.xml"
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< com.google.cloud.tools:dashboard >------------------
[INFO] Building Cloud Tools Open Source Code Hygiene Dashboard 1.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:1.6.0:java (default-cli) @ dashboard ---
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[WARNING] 
org.eclipse.aether.resolution.DependencyResolutionException: Could not find artifact io.grpc:protoc-gen-grpc-java:jar:1.23.0 in central (http://repo1.maven.org/maven2/)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies (DefaultRepositorySystem.java:352)
    at com.google.cloud.tools.opensource.dependencies.DependencyGraphBuilder.resolveCompileTimeDependencies (DependencyGraphBuilder.java:162)
    at com.google.cloud.tools.opensource.dependencies.DependencyGraphBuilder.getStaticLinkageCheckDependencyGraph (DependencyGraphBuilder.java:191)
    at com.google.cloud.tools.opensource.classpath.ClassPathBuilder.artifactsToDependencyPaths (ClassPathBuilder.java:86)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.generate (DashboardMain.java:182)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.generate (DashboardMain.java:171)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.main (DashboardMain.java:108)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:282)
    at java.lang.Thread.run (Thread.java:748)
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not find artifact io.grpc:protoc-gen-grpc-java:jar:1.23.0 in central (http://repo1.maven.org/maven2/)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:423)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:225)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies (DefaultRepositorySystem.java:335)
    at com.google.cloud.tools.opensource.dependencies.DependencyGraphBuilder.resolveCompileTimeDependencies (DependencyGraphBuilder.java:162)
    at com.google.cloud.tools.opensource.dependencies.DependencyGraphBuilder.getStaticLinkageCheckDependencyGraph (DependencyGraphBuilder.java:191)
    at com.google.cloud.tools.opensource.classpath.ClassPathBuilder.artifactsToDependencyPaths (ClassPathBuilder.java:86)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.generate (DashboardMain.java:182)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.generate (DashboardMain.java:171)
    at com.google.cloud.tools.opensource.dashboard.DashboardMain.main (DashboardMain.java:108)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:282)
    at java.lang.Thread.run (Thread.java:748)
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact io.grpc:protoc-gen-grpc-java:jar:1.23.0 in central (http://repo1.maven.org/maven2/)
    at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed (ArtifactTransportListener.java:48)
    at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run (BasicRepositoryConnector.java:369)
    at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run (RunnableErrorForwarder.java:75)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)
```

The problem is the artifact packaging is pom while our dashboard is trying to download a JAR file.

**Comment by suztomo** [comment]
Memo:

grpc-bom has the artifact without specifying `packaging:pom`:

https://search.maven.org/artifact/io.grpc/grpc-bom/1.23.0/pom

```
      <dependency>
        <groupId>io.grpc</groupId>
        <artifactId>protoc-gen-grpc-java</artifactId>
        <version>1.23.0</version>
      </dependency>
```

Is this intentional? To ask gRPC team.

**Comment by elharo** [comment]
This was fixed here: https://github.com/grpc/grpc-java/commit/a8137dc18f39151d2e0addb61735bb7579d2062e#diff-114533b5cf32ea0b42db258675e859db

This should roll out with the next release, 1.25.0 probably.

**Comment by suztomo** [comment]
Fixed at 1.24.1
